src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestRestController.java

```diff
@@ -0,0 +1,247 @@
+import org.openelisglobal.common.util.ConfigurationProperties;
+import org.openelisglobal.common.util.IdValuePair;
+import org.openelisglobal.common.util.ConfigurationProperties.Property;
+import org.openelisglobal.internationalization.MessageUtil;
+import org.openelisglobal.observationhistory.service.ObservationHistoryService;
+import org.openelisglobal.observationhistory.service.ObservationHistoryServiceImpl.ObservationType;
+import org.openelisglobal.patient.service.PatientService;
+import org.openelisglobal.person.service.PersonService;
+import org.openelisglobal.result.action.util.ResultsLoadUtility;
+import org.openelisglobal.role.service.RoleService;
+import org.openelisglobal.sample.service.SampleService;
+import org.openelisglobal.sample.valueholder.Sample;
+import org.openelisglobal.samplehuman.service.SampleHumanService;
+import org.openelisglobal.sampleqaevent.service.SampleQaEventService;
+import org.openelisglobal.sampleqaevent.valueholder.SampleQaEvent;
+import org.openelisglobal.search.service.SearchResultsService;
+import org.openelisglobal.spring.util.SpringContext;
+import org.openelisglobal.systemuser.service.UserService;
+import org.openelisglobal.test.beanItems.TestResultItem;
+import org.openelisglobal.test.service.TestSectionService;
+import org.openelisglobal.test.service.TestService;
+import org.openelisglobal.test.service.TestServiceImpl;
+import org.openelisglobal.workplan.controller.WorkplanRestController.ValueComparator;
+import org.openelisglobal.workplan.form.WorkplanForm;
+import org.openelisglobal.workplan.form.WorkplanForm.PrintWorkplan;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.ModelAndView;
+
+@RestController
+@RequestMapping(value = "/rest/")
+public class WorkplanByTestRestController extends WorkplanRestController {
+
+	@Autowired
+	private AnalysisService analysisService;
+	@Autowired
+	private SampleQaEventService sampleQaEventService;
+	@Autowired
+	private UserService userService;
+
+	private static boolean HAS_NFS_PANEL = false;
+
+	static {
+		HAS_NFS_PANEL = ConfigurationProperties.getInstance().isPropertyValueEqual(Property.CONDENSE_NFS_PANEL, "true");
+	}
+
+	@GetMapping(value = "test-list", produces = MediaType.APPLICATION_JSON_VALUE)
+	@ResponseBody
+	private List<IdValuePair> getTestDropdownList(HttpServletRequest request) {
```

Comment:
can you put thiese common Lists under [common](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/2.8/src/main/java/org/openelisglobal/common/rest/DisplayListController.java)

Reply:
solved